pkg/operator/controller/dns/controller.go

```diff
@@ -390,6 +427,35 @@ func recordIsAlreadyPublishedToZone(record *iov1.DNSRecord, zoneToPublish *confi
 	return false
 }
 
+// domainIsAlreadyPublishedToZone returns true if the domain name in the provided DNSRecord is already published by
+// another existing dnsRecord.
+func domainIsAlreadyPublishedToZone(ctx context.Context, cache cache.Cache, record *iov1.DNSRecord, zone *configv1.DNSZone) (bool, error) {
+	records := iov1.DNSRecordList{}
+	if err := cache.List(ctx, &records, client.MatchingFields{dnsRecordIndexFieldName: record.Spec.DNSName}); err != nil {
+		return false, err
+	}
+
+	if len(records.Items) == 0 {
+		return false, nil
+	}
+
+	for _, existingRecord := range records.Items {
+		// we only care if the domain name is published by a different record, so ignore the matching record if it
+		// already exists.
+		// TODO: There's got to be a better way to match the same object
```

Comment:
You can match the UID here. I would say that during admission webhooks matching UID is a bad idea, as the UID will be added once persisted by apiserver, but in case of a reconciliation process it means that the UID was already generated and cannot be changed anymore.